### PR TITLE
Moar packages and configs 

### PR DESCRIPTION
**Pull request by Xepe**
Add some packages for common use in different languages 

Also I started to use fzf instead of ctrl p but I still need ctrl because vim-bookmarks that it is pretty cool

### CODE REVIEW

File: .vimrc [hunk 1]
```diff
@@ -57,7 +65,7 @@ call plug#end()
 " Performance boosters
 :set nocursorline
 :set lazyredraw
-:set re=1
+:set re=0
```
Comment (Xepe) [inline .vimrc:68]: I should rever this change and find a way to make it work with jsx and tsx files

File: .vimrc [hunk 2]
```diff
@@ -202,8 +211,8 @@ noremap L g_
 :let g:coc_global_extensions=[
             \'coc-diagnostic',
             \'coc-marketplace',
-            \'coc-tabnine',
             \'coc-snippets',
+            \'coc-tabnine',
```
Comment (Xepe) [inline .vimrc:215]: useful but does it worth it ?  @radyz  what do you think ?

File: .vimrc [hunk 3]
```diff
@@ -107,6 +115,7 @@ autocmd GUIEnter * set visualbell t_vb=
 :set t_Co=256
 :set background=dark
 :colorscheme hybrid_reverse
+:set synmaxcol=0
```
Comment (Xepe) [inline .vimrc:118]: This  fix the highlight when we have long lines :) 

cc: @radyz

File: .vimrc [hunk 4]
```diff
@@ -354,6 +364,7 @@ set ttimeoutlen=50
 " NERDTREE
 " ============================================================================
 :nnoremap <Leader>nt :NERDTreeToggle<CR>
+:nnoremap <Leader>nf :NERDTreeFind<CR>
```
Comment (Xepe) [inline .vimrc:367]: I highlight recommend this one when you have a huge tree of folder <3

File: .vimrc [hunk 5]
```diff
@@ -420,6 +434,8 @@ hi illuminatedWord cterm=underline gui=underline
 " ============================================================================
 " BETTER COMMENTS 
 " ============================================================================
+let g:bettercomments_skipped = ['python']
```
Comment (Xepe) [inline .vimrc:437]: I need to review this change @radyz  probably is not need it now. I think that I added it because it breaks formatter or the code highlighting

File: coc-settings.json [hunk 6]
```diff
@@ -23,12 +25,19 @@
...
-                "/usr/lib/dart/bin/snapshots/analysis_server.dart.snapshot",
+                "/usr/local/opt/dart/libexec/bin/snapshots/analysis_server.dart.snapshot",
                 "--lsp"
             ],
             "filetypes": ["dart"],
             "disableDynamicRegister": true,
             "trace.server": "verbose"
+        },
+        "terraform": {
+            "command": "terraform-lsp",
```
Comment (Xepe) [inline coc-settings.json:36]: Maybe I could remove it I work with terraform that is why it is there but it is really specif and also we need to install the terraform-lsp so I am not sure

File: coc-settings.json [hunk 7]
```diff
@@ -39,9 +48,16 @@
     "python.linting.enabled": true,
     "python.linting.pylintEnabled": false,
     "python.linting.flake8Enabled": true,
+    "python.linting.flake8Args": ["--max-line-length", "88"],
```
Comment (Xepe) [inline coc-settings.json:51]: Sorry but not sorry 🕺🏻

File: .vimrc (hunk 2 above)
Comment (radyz) [reply .vimrc:215]: I'd keep it. Suggestions work great when no actual intellisense is available

File: .vimrc [hunk 8]
```diff
@@ -404,7 +417,8 @@ autocmd VimEnter * if empty(expand('<amatch>'))|call FugitiveDetect(getcwd())|en
 " Signify
 nmap <leader>gj <plug>(signify-next-hunk)
 nmap <leader>gk <plug>(signify-prev-hunk)
-:nnoremap <Leader>gh :SignifyHunkDiff<CR>
+:nnoremap <Leader>ghd :SignifyHunkDiff<CR>
```
Comment (radyz) [inline .vimrc:420]: Why was this changed?

File: .vimrc [hunk 9]
```diff
@@ -436,3 +452,33 @@ let g:matchup_matchparen_deferred = 1
...
+" ============================================================================
+" ANYFOLDS
+" ============================================================================
+autocmd Filetype * AnyFoldActivate   
+set foldlevel=99 " Open all folds
+
+" ============================================================================
+" LENGTHMATTERS
+" ============================================================================
+let g:lengthmatters_start_at_column = 88
```
Comment (radyz) [inline .vimrc:465]: This should be in a local file as this depends per project/user.

File: .vimrc [hunk 10]
```diff
@@ -436,3 +452,33 @@ let g:matchup_matchparen_deferred = 1
...
+"nnoremap <silent> <space>f :exe 'Files ' . <SID>fzf_root()<CR>
+nnoremap <silent> <space>f :call fzf#run(fzf#vim#with_preview(fzf#wrap({'source': 'ag --hidden --ignore node_modules --ignore .git -g ""', 'dir': <SID>fzf_root()})))<CR>
+nnoremap <silent> <space><space> :call fzf#run(fzf#vim#with_preview(fzf#wrap({'source': 'ag --hidden --ignore node_modules --ignore .git -g ""'})))<CR>
+nnoremap <silent> <space>c :CocFzfList<CR>
+
+" ============================================================================
+" COC PYTHON 
+" ============================================================================
+
+call coc#config('python', {'pythonPath': split(execute('!which python'), '\n')[-1]})
```
Comment (radyz) [inline .vimrc:484]: What does this do?

File: coc-settings.json [hunk 11]
```diff
@@ -23,12 +25,19 @@
         "dart": {
             "command": "dart",
             "args": [
-                "/usr/lib/dart/bin/snapshots/analysis_server.dart.snapshot",
+                "/usr/local/opt/dart/libexec/bin/snapshots/analysis_server.dart.snapshot",
```
Comment (radyz) [inline coc-settings.json:28]: This should probably be configured somewhere as it looks it depends on user specific paths

File: coc-settings.json (hunk 7 above)
Comment (radyz) [reply coc-settings.json:51]: Why would you need this? `flake8` should be read from config file instead using `.flake8`

This way it can be overriden per project/user

https://flake8.pycqa.org/en/latest/user/configuration.html#configuration-locations

File: coc-settings.json [hunk 12]
```diff
@@ -39,9 +48,16 @@
     "python.linting.enabled": true,
     "python.linting.pylintEnabled": false,
     "python.linting.flake8Enabled": true,
+    "python.linting.flake8Args": ["--max-line-length", "88"],
     "python.analysis.logLevel": "Warning",
     "python.formatting.provider": "black",
-    "python.formatting.blackArgs": ["--line-length", "80"],
+    "python.formatting.autopep8Args": [
```
Comment (radyz) [inline coc-settings.json:54]: This should be in a user configurable setting (preferably in a project `.coc-settings.json` file. Default is black, but changing it to `autopep8` might make sense in other projects or not.

**Comment by radyz** [comment]
I'd like to integrate something like this: https://www.vim.org/scripts/script.php?script_id=441, so we have a set of defaults and then each user/project can override them as not every[one/thing] will be the same each time. This way we give the users the option to change their configs from the defaults based on preferences, projects or even provide defined templates

File: .vimrc (hunk 8 above)
Comment (Xepe) [reply .vimrc:420]: because I added a new map for `SignifyHunkUndo` so it not make sense only to have gh for `SignifyHunkDiff` now

File: .vimrc (hunk 9 above)
Comment (Xepe) [reply .vimrc:465]: I am not sure, since I would like to have it as default length for the team

File: .vimrc (hunk 10 above)
Comment (Xepe) [reply .vimrc:484]: there is a bug with coc-python it does not pic the correct interpreter when you are working with multiple envs, with this config we ensure that we are suing the correct python interpreter by env

File: coc-settings.json (hunk 11 above)
Comment (Xepe) [reply coc-settings.json:28]: yes, I think so 🤔  this kinds of config are very specific maybe we should not added it in here for terraform and dart I mean

File: coc-settings.json (hunk 12 above)
Comment (Xepe) [reply coc-settings.json:54]: same as before, I did it here because I like the idea of use as default 88 🤔  as default for the projects but yeah we could add it in .coc-settings.json by project

**Comment by Xepe** [comment]
I am going split the PR in small units
